### PR TITLE
Set region in sts client creation also

### PR DESCRIPTION
**Pull request by damianxu88**

### CODE REVIEW

**Comment by damianxu88** [comment]
@realstraw @realstraw Can I get a review for this PR?

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/Client.scala [hunk 1]
```diff
@@ -34,9 +34,13 @@ object Client {
       awsAssumeRoleArn <- ProviderSettings().awsAssumeRoleArn
     } yield {
 
-      val stsClient = staticCredentialsOpt.map { staticCredentials =>
-        StsClient.builder().credentialsProvider(staticCredentials).build()
-      }.getOrElse(StsClient.builder().build())
+      val stsClientBuilder = clientRegionOpt
+        .fold(StsClient.builder())(StsClient.builder().region)
```
Comment (realstraw) [inline orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/Client.scala:38]: `foldLeft` is better in this case

Comment (damianxu88) [reply orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/Client.scala:38]: @realstraw let me update

Comment (damianxu88) [reply orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/Client.scala]: @realstraw ok, changed

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/Client.scala [hunk 2]
```diff
@@ -34,9 +34,15 @@ object Client {
       awsAssumeRoleArn <- ProviderSettings().awsAssumeRoleArn
     } yield {
 
-      val stsClient = staticCredentialsOpt.map { staticCredentials =>
-        StsClient.builder().credentialsProvider(staticCredentials).build()
-      }.getOrElse(StsClient.builder().build())
+      val stsClientBuilder = clientRegionOpt
+        .foldLeft(StsClient.builder())((builder, region) =>
+          builder.region(region)
```
Comment (realstraw) [inline orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/Client.scala:39]: `_.region(_)` is shorter

Comment (damianxu88) [reply orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/Client.scala]: @realstraw ok, changed

**Review by realstraw — APPROVED** [review]

**Pull request merged**